packages/trader/src/AppV2/Hooks/useActiveSymbols.ts

```diff
@@ -30,89 +30,133 @@ const useActiveSymbols = () => {
         onChange,
         setActiveSymbolsV2,
         symbol,
-        is_trade_component_mounted,
     } = useTraderStore();
+    const [activeSymbols, setActiveSymbols] = useState<ActiveSymbols | []>(symbols_from_store);
+
+    const { available_contract_types, is_fetching_ref: is_contracts_loading_ref } = useContractsForCompany();
 
     const default_symbol_ref = useRef('');
-    const previous_logged_in = usePrevious(is_logged_in);
     const previous_contract_type = usePrevious(contract_type);
+    const prev_loginid = useRef(loginid);
+
+    const trade_types_with_barrier_category = [
+        TRADE_TYPES.RISE_FALL,
+        TRADE_TYPES.RISE_FALL_EQUAL,
+        TRADE_TYPES.HIGH_LOW,
+    ] as string[];
 
     const getContractTypesList = () => {
         if (is_turbos) return [CONTRACT_TYPES.TURBOS.LONG, CONTRACT_TYPES.TURBOS.SHORT];
         if (is_vanilla) return [CONTRACT_TYPES.VANILLA.CALL, CONTRACT_TYPES.VANILLA.PUT];
         return getContractTypesConfig()[contract_type]?.trade_types ?? [];
     };
 
-    const fetchActiveSymbols = useCallback(
-        async () => {
-            let response;
-
-            const trade_types_with_barrier_category = [
-                TRADE_TYPES.RISE_FALL,
-                TRADE_TYPES.RISE_FALL_EQUAL,
-                TRADE_TYPES.HIGH_LOW,
-            ] as string[];
-            const barrier_category = ContractType.getBarrierCategory(contract_type).barrier_category;
-
-            const request = {
-                active_symbols: 'brief',
-                contract_type: getContractTypesList(),
-                ...(trade_types_with_barrier_category.includes(contract_type) && barrier_category
-                    ? { barrier_category: [barrier_category] }
-                    : {}),
+    const { barrier_category } = (available_contract_types?.[contract_type]?.config || {}) as any;
+
+    const isQueryEnabled = useCallback(() => {
+        if (!available_contract_types) return false;
+        if (is_contracts_loading_ref.current) return false;
+        return true;
```

Comment:
Maybe something like this?
` return (!available_contract_types || is_contracts_loading_ref.current) ? false: true;`

Reply:
Updated it. Still want to keep it as 2 lines to make the debugging easier.